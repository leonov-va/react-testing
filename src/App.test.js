import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("TEST APP", () => {
  test("getByText, getByRole, getByPlaceholderText", () => {
    render(<App />);
    const helloWorldElem = screen.getByText(/hello world/i);
    const btn = screen.getByRole("button");
    const input = screen.getByPlaceholderText(/input value/i);

    expect(helloWorldElem).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();

    // screen.debug()
  });

  test("queryByText (hello2)", () => {
    render(<App />);
    const helloWorldElem = screen.queryByText(/hello2/i);
    expect(helloWorldElem).toBeNull();
  });

  test("async findByText", async () => {
    render(<App />);
    // screen.debug()
    const helloWorldElem = await screen.findByText(/data/i);
    expect(helloWorldElem).toBeInTheDocument();
    expect(helloWorldElem).toHaveStyle({ color: "red" });
    // screen.debug()
  });

  test("click event", async () => {
    render(<App />);
    const btn = screen.getByTestId("toggle-btn");
    expect(screen.queryByTestId("toggle-elem")).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle-elem")).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle-elem")).toBeNull();
  });

  test("input event", async () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/input value/i);
    expect(screen.queryByTestId("value-elem")).toContainHTML("");
    // Искуственное событие
    // fireEvent.input(input, {
    //   target: {
    //     value: "123123",
    //   },
    // });
    // Близко к пользователю, обрабатываются события нажатия клавиш и тд
    userEvent.type(input, "123123");
    expect(screen.queryByTestId("value-elem")).toContainHTML("123123");
  });
});
