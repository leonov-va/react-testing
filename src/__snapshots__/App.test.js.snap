// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP getByText, getByRole, getByPlaceholderText 1`] = `
<input
  placeholder="input value...."
  type="text"
/>
`;

exports[`TEST APP renders hello world, button, input 1`] = `
<input
  placeholder="input value...."
  type="text"
/>
`;

exports[`TEST APP renders learn hello world 1`] = `
<input
  placeholder="input value...."
  type="text"
/>
`;

exports[`TEST APP renders learn react link 1`] = `
<input
  placeholder="input value...."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value...."
  type="text"
/>
`;
